### ISSUE TITLE
Exception: EXCEPTION_ACCESS_VIOLATION

### ISSUE THREAD

**Issue by xtk923** [JuliaDatabases/MySQL.jl#129]
Hi,

I'm trying to insert more than a few millions of rows into a MySQL table, using MySQL.jl. And I frequently run into this exception at some point in the for-loop with `MySQL.execute!(stmt, vals)`

Here's the error code I receive:
`Please submit a bug report with steps to reproduce this fault, and any error messages that follow (in their entirety). Thanks.
Exception: EXCEPTION_ACCESS_VIOLATION at 0x7ffeb250db80 -- mysql_change_user_cont at C:\Users\username\.julia\packages\MySQL\owtkt\deps\usr\bin\libmariadb.dll (unknown line)
in expression starting at D:\DatabasePlaygroud\create_DB.jl:95
mysql_change_user_cont at C:\Users\username\.julia\packages\MySQL\owtkt\deps\usr\bin\libmariadb.dll (unknown line)
mysql_change_user_cont at C:\Users\username\.julia\packages\MySQL\owtkt\deps\usr\bin\libmariadb.dll (unknown line)
mysql_change_user_cont at C:\Users\username\.julia\packages\MySQL\owtkt\deps\usr\bin\libmariadb.dll (unknown line)
mysql_stmt_execute at C:\Users\username\.julia\packages\MySQL\owtkt\src\api.jl:185 [inlined]
execute! at C:\Users\username\.julia\packages\MySQL\owtkt\src\prepared.jl:47
unknown function (ip: 000000002AE77D2B)
jl_apply_generic at /home/Administrator/buildbot/worker/package_win64/build/src/home/Administrator/buildbot/worker/package_win64/build/src\gf.c:2197
macro expansion at D:\DatabasePlaygroud\create_DB.jl:76 [inlined]
macro expansion at C:\Users\username\.julia\packages\ProgressMeter\NIpPa\src\ProgressMeter.jl:606 [inlined]
macro expansion at D:\DatabasePlaygroud\create_DB.jl:60 [inlined]
macro expansion at .\util.jl:156 [inlined]
to_DB at D:\DatabasePlaygroud\create_DB.jl:49
unknown function (ip: 000000002AE6804E)
jl_fptr_trampoline at /home/Administrator/buildbot/worker/package_win64/build/src/home/Administrator/buildbot/worker/package_win64/build/src\gf.c:1842
jl_apply_generic at /home/Administrator/buildbot/worker/package_win64/build/src/home/Administrator/buildbot/worker/package_win64/build/src\gf.c:2197
do_call at /home/Administrator/buildbot/worker/package_win64/build/src/home/Administrator/buildbot/worker/package_win64/build/src\interpreter.c:323
eval_value at /home/Administrator/buildbot/worker/package_win64/build/src/home/Administrator/buildbot/worker/package_win64/build/src\interpreter.c:411
eval_stmt_value at /home/Administrator/buildbot/worker/package_win64/build/src/home/Administrator/buildbot/worker/package_win64/build/src\interpreter.c:362 [inlined]
eval_body at /home/Administrator/buildbot/worker/package_win64/build/src/home/Administrator/buildbot/worker/package_win64/build/src\interpreter.c:759
jl_interpret_toplevel_thunk_callback at /home/Administrator/buildbot/worker/package_win64/build/src/home/Administrator/buildbot/worker/package_win64/build/src\interpreter.c:885
unknown function (ip: FFFFFFFFFFFFFFFE)
unknown function (ip: 0000000011F3559F)
unknown function (ip: 0000000000000001)
jl_toplevel_eval_flex at /home/Administrator/buildbot/worker/package_win64/build/src/home/Administrator/buildbot/worker/package_win64/build/src\toplevel.c:764
jl_parse_eval_all at /home/Administrator/buildbot/worker/package_win64/build/src/home/Administrator/buildbot/worker/package_win64/build/src\ast.c:883
jl_load at /home/Administrator/buildbot/worker/package_win64/build/src/home/Administrator/buildbot/worker/package_win64/build/src\toplevel.c:826 [inlined]
jl_load_ at /home/Administrator/buildbot/worker/package_win64/build/src/home/Administrator/buildbot/worker/package_win64/build/src\toplevel.c:833
include at .\boot.jl:326 [inlined]
include_relative at .\loading.jl:1038
include at .\sysimg.jl:29
jl_apply_generic at /home/Administrator/buildbot/worker/package_win64/build/src/home/Administrator/buildbot/worker/package_win64/build/src\gf.c:2197
exec_options at .\client.jl:267
_start at .\client.jl:436
jl_apply_generic at /home/Administrator/buildbot/worker/package_win64/build/src/home/Administrator/buildbot/worker/package_win64/build/src\gf.c:2197
jl_apply at /home/Administrator/buildbot/worker/package_win64/build/ui/home/Administrator/buildbot/worker/package_win64/build/src\julia.h:1571 [inlined]
true_main at /home/Administrator/buildbot/worker/package_win64/build/ui/home/Administrator/buildbot/worker/package_win64/build/ui\repl.c:96
wmain at /home/Administrator/buildbot/worker/package_win64/build/ui/home/Administrator/buildbot/worker/package_win64/build/ui\repl.c:217
__tmainCRTStartup at /usr/src/debug/mingw64-x86_64-runtime-6.0.0-1/usr/src/debug/mingw64-x86_64-runtime-6.0.0-1/crt\crtexe.c:334
mainCRTStartup at /usr/src/debug/mingw64-x86_64-runtime-6.0.0-1/usr/src/debug/mingw64-x86_64-runtime-6.0.0-1/crt\crtexe.c:223
BaseThreadInitThunk at C:\WINDOWS\System32\KERNEL32.DLL (unknown line)
RtlUserThreadStart at C:\WINDOWS\SYSTEM32\ntdll.dll (unknown line)
Allocations: 102238317 (Pool: 102112515; Big: 125802); GC: 444`

The part of code for inserting the data is pasted as follows:

        df = DataFrame(<some data frame of size = (400000, 63))
        valuerow = "(" * (["?" for x = 1:63] |> x->join(x, ",")) * ")"
        MySQL.execute!(cnx, "SET autocommit=0")
        @time begin
            in_batch = true

            if in_batch
                batch_size = 10
                num_batches = ceil(nrow(df)/batch_size) |> Int
                stmt = MySQL.Stmt(cnx, "INSERT IGNORE INTO cumulative_pd (U3_ID, Data_Date," *
                                  sixty_monthStr *
                                  ", Update_Time) VALUES " *
                                  join([valuerow for x = 1:batch_size], ", ") * ";")

                @showprogress for b = 1:num_batches
                    b_s = (b -1) * batch_size + 1
                    b_e = min(nrow(df), b * batch_size)
                    this_batch_size = b_e - b_s + 1
                    if b == num_batches
                        stmt = MySQL.Stmt(cnx, "INSERT IGNORE INTO cumulative_pd (U3_ID, Data_Date," *
                                          sixty_monthStr *
                                          ", Update_Time) VALUES " *
                                          join([valuerow for x = 1:this_batch_size], ", ") * ";")
                    end

                    # insert data
                    begin
                        vals = eachrow(df[b_s:b_e, :]) |> x->Vector.(x) |> x->vcat(x...)
                        sleep(0.001)
                        # MySQL.execute!(df, stmt)
                        MySQL.execute!(stmt, vals)
                    end

                end
            else
                stmt = MySQL.Stmt(cnx, "INSERT IGNORE INTO cumulative_pd (U3_ID, Data_Date," *
                                  sixty_monthStr *
                                  ", Update_Time) VALUES " * valuerow * ";")
                MySQL.execute!(df, stmt)
            end

        end
        MySQL.execute!(cnx, "COMMIT;")

Basically I turn off `autocommit` and make a bunch of inserts, then `commit` them. I also tried to use `MySQL.execute!(df, stmt)`, but the same problem occurs at some point into the loop.

Would you have any suggestions about what could have caused this? Thanks!

**Comment by quinnj** [comment]
The query execution and prepared statements have been overhauled to be more correct and safe in https://github.com/JuliaDatabases/MySQL.jl/pull/139, so it will be worth seeing if you get the same issue once that's released.
